### PR TITLE
[#5107] Added test to verify istream supports replica numbers when writing (master)

### PR DESCRIPTION
**Pull request by korydraughn**

### CODE REVIEW

File: scripts/irods/test/test_istream.py [hunk 1]
```diff
@@ -35,3 +36,47 @@ def test_istream_empty_file(self):
...
+    def test_creating_data_objects_using_replica_numbers_is_not_allowed__issue_5107(self):
+        data_object = 'cannot_create_data_objects'
+
+        # Show that the targeted data object does not exist.
+        self.admin.assert_icommand(['ils', '-l', data_object], 'STDERR', ['does not exist or user lacks access permission'])
+
+        # Show that attempts to write to a non-existent data object using replica numbers does
+        # not result in creation of the data object.
+        self.admin.assert_icommand_fail(['istream', 'write', '-n0', data_object], input='some data')
+        self.admin.assert_icommand_fail(['istream', 'write', '-n4', data_object], input='some data')
```
Comment (trel) [inline scripts/irods/test/test_istream.py:49]: assert the specific error seen?

Comment (korydraughn) [reply scripts/irods/test/test_istream.py:49]: not sure how to do that with the `input=` parameter, but will investigate.

**Comment by trel** [comment]
please add another fail test for istream to write to non-existent replica of existing data object

File: scripts/irods/test/test_istream.py (hunk 1 above)
Comment (trel) [reply scripts/irods/test/test_istream.py:49]: ah, gotcha.

**Comment by korydraughn** [comment]
test added.

added #

**Pull request merged**